tests/testutil/service.go

```diff
@@ -388,12 +533,30 @@ func (ts *TestService) waitForReadyService() error {
 	return nil
 }
 
+// Wait until a service has finished building, or error out
+func (ts *TestService) waitForBuild() error {
+	f := wait.ConditionFunc(func() (bool, error) {
+		err := ts.reloadBuild()
+		if err == nil {
+			if ts.Build.Status.GetCondition(apis.ConditionSucceeded) != nil && ts.Build.Status.GetCondition(apis.ConditionSucceeded).IsTrue() {
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+	err := wait.Poll(10*time.Second, 240*time.Second, f)
+	if err != nil {
+		return fmt.Errorf("build never completed for service: %v. Error: %v", ts.Name, err.Error())
+	}
+	return nil
+}
+
 // Wait until a service hits wanted number of available replicas, or error out
 func (ts *TestService) waitForAvailableReplicas(want int) error {
 	f := wait.ConditionFunc(func() (bool, error) {
 		err := ts.reload()
 		if err == nil {
-			if ts.GetAvailableReplicas() == want {
+			if ts.Service.Status.DeploymentStatus != nil && int(ts.Service.Status.DeploymentStatus.AvailableReplicas) == want {
```

Comment:
Whats point of this change ? Isn't that exactly what `GetAvailableReplicas` does ?

Reply:
I changed this like 8 times trying to solve flakiness. This was a mistake :(